pkg/kube/client.go

```diff
@@ -399,14 +399,27 @@ func createPatch(mapping *meta.RESTMapping, target, current runtime.Object) ([]b
 		return nil, types.StrategicMergePatchType, fmt.Errorf("serializing target configuration: %s", err)
 	}
 
+	// Get a versioned object
+	versionedObject, err := mapping.ConvertToVersion(target, mapping.GroupVersionKind.GroupVersion())
+
+	// Unstructured objects, such as CRDs, may not have an not registered error
+	// returned from ConvertToVersion. Anything that's unstructured should
+	// use the jsonpatch.CreateMergePatch. Strategic Merge Patch is not supported
+	// on objects like CRDs.
+	_, isUnstructured := versionedObject.(runtime.Unstructured)
+
 	if apiequality.Semantic.DeepEqual(oldData, newData) {
+		// Unstructured types, such as CRDs, are not supported by Strategic
+		// Merge Patches
+		if isUnstructured {
+			return nil, types.MergePatchType, nil
```

Comment:
The line below is already returning a NOP.  This can be deleted.

Reply:
I'll work on an update. Thanks.